Optional /whitelist args - sorted, nobase

ecoppen opened the pull request:
## Summary
Added two optional arguments to `/whitelist` command in Telegram:

- `sorted` for alphabetical order
- `nobase` to remove the base currency

## Quick changelog
- `/whitelist sorted` for alphabetical whitelist
- `/whitelist nobase` to remove the base currency from each pairing in the whitelist
- `/whitelist sorted nobase` for an alphabetical whitelist without base currencies listed

## Other notes
I gave the testing a go but it might need further adjustment? 😅

## Example 
<img width="359" alt="image" src="https://user-images.githubusercontent.com/51025241/183921077-89e347d7-0098-4fa1-8da0-61da5506ce7d.png">

ecoppen committed:
Optional /whitelist args - sorted, nobase

Added two optional arguments for whitelist - `sorted` for alphabetical order and `nobase` for displaying the whitelist without base currency e.g. /USDT.

Updated help with optional commands.

Added a space in an unrelated help message.

ecoppen committed:
Update tests for sorted and nobase

Tests for PR #7211

xmatthias commented on tests/rpc/test_rpc_telegram.py:
```diff
@@ -1458,6 +1458,23 @@ def test_whitelist_static(default_conf, update, mocker) -> None:
+    context = MagicMock()
+    context.args = ['nobase', 'sorted']
+    msg_mock.reset_mock()
+    assert ("Using whitelist `['StaticPairList']` with 4 pairs\n"
```
You'll have to call the whitelist again.
just resetting the mock - and then asserting won't work properly (and is causing the ci failure)

applies to each of the asserts above.

ecoppen replied:
Thanks for this guidance, this is the area I'm least confident with so should be resolved now if it's as simple as I understood it to be.

xmatthias commented on freqtrade/rpc/telegram.py:
```diff
@@ -1368,6 +1368,12 @@ def _whitelist(self, update: Update, context: CallbackContext) -> None:
+            if context.args:
+                if "sorted" in context.args:
+                    whitelist['whitelist'] = sorted(whitelist['whitelist'])
+                if "nobase" in context.args:
+                    whitelist['whitelist'] = [pair.split("/")[0] for pair in whitelist['whitelist']]
```
A currency pair has the following naming: `base/quote[:settle]` (settle only for futures).

Keeping the above in mind, think the selected naming is pretty confusing.

it's called nobase - but in reality, it's removing the quote currency (and potentially the settle currency), leaving the base-currency in place, so either "baseonly" or "noquote" would be more fitting (i'd prefer the positive version though, as the only thing left will be the base).

ecoppen replied:
I can't believe I've mixed these two up, especially after referring to them in the correct way on discord, perhaps I second guessed myself. I have changed to `baseonly` following your recommendation.

xmatthias reviewed (changes requested):
Looks great and works as expected (even though CI still disagrees, the problem there is really with the test).

Please also add the new comands to the `telegram.md` documentation page - so people can also find out about the new commands.

ecoppen committed:
nobase -> baseonly

ecoppen committed:
fix - reload whitelist

Should fix the issue, if not I'll move development to a different computer and get local testing running properly.

ecoppen committed:
Update telegram-usage.md

Add the optional arguments to the documentation.

ecoppen commented:
> Please also add the new comands to the `telegram.md` documentation page - so people can also find out about the new commands.

Hopefully this is `telegram_usage.md` otherwise I've updated the wrong one.

xmatthias committed:
Fix broken telegram tests

ecoppen commented:
Thanks for fixing the tests!

xmatthias reviewed (approved):
Tests are ... often confusing for a lot of people - and it's no problem to ask for help, considering an attempt was made :1st_place_medal: 

LGTM :+1: 

Thanks for your contribution! :100:
